### PR TITLE
Don't allow field extractors to match field name suffixes

### PR DESCRIPTION
**Pull request by lava**
###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

**Comment by lava** [comment]
After thinking about this PR a bit, I think this might break queries like `zeek.* == "xxx"`. Since there's no unit test for these and they would have only very rarely worked previously, I'm not sure if this was supposed to work. But as a user, it seems like something I'd intuitively expect to work, if wildcards are supported at all.

**Comment by tobim** [comment]
> After thinking about this PR a bit, I think this might break queries like `zeek.* == "xxx"`. Since there's no unit test for these and they would have only very rarely worked previously, I'm not sure if this was supposed to work. But as a user, it seems like something I'd intuitively expect to work, if wildcards are supported at all.

The bug that this PR fixes prevented those queries to work as well, so I don't think there is any harm done.

**Comment by mavam** [comment]
> But as a user, it seems like something I'd intuitively expect to work, if wildcards are supported at all.

Yeah, I agree. let's treat this as a separate feature that we don't support yet.

If we stick with component-based suffix matching, we're good for this PR.

**Review by tobim — CHANGES_REQUESTED** [review]
This is still missing a changelog entry, otherwise it is good to go.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ This changelog documents all notable user-facing changes of VAST.
 
 ## Unreleased
 
+- 🐞 Partitions containing multiple fields with the same field name suffix could
+  fail to return some query results due to a bug in the type resolver.
+  [#1447](https://github.com/tenzir/vast/pull/1447)
```
Comment (tobim) [inline CHANGELOG.md:18]: This is very technical, the changelog should be more geared towards the user.
How about:
```
A query for a field or field name suffix that matches multiple fields of different
types would erroneously return no results.
```

Comment (lava) [reply CHANGELOG.md:18]: I think we agreed not to do changelog wordsmithing in the individual PRs in favor of one big wording pass when we rearrange the changlog for the release.

Comment (dominiklohmann) [reply CHANGELOG.md:18]: To be fair, that's not really wordsmithing but rather changing the content entirely.

**Review by tobim — APPROVED** [review]

**Pull request merged**